physlr/physlr.py

```diff
@@ -1674,62 +1799,141 @@ def physlr_molecules(self):
-            file=sys.stderr)
-
-        # Add edges.
-        for (u, v), prop in gin.edges.items():
-            # Skip singleton and cut vertices, which are excluded from the partition.
-            if v not in molecules[u] or u not in molecules[v]:
-                continue
-            u_molecule = molecules[u][v]
-            v_molecule = molecules[v][u]
-            gout.add_edge(f"{u}_{u_molecule}", f"{v}_{v_molecule}", m=prop["m"])
-        print(int(timeit.default_timer() - t0), "Separated molecules", file=sys.stderr)
+            # Add edges
+            if round > 1:
+                for u, vs in sorted(molecules.items()):
+                    for v, _ in vs.items():
+                        if v in molecules:
+                            if v not in molecules[u] or u not in molecules[v]:
+                                continue
+                            u_molecule = molecules[u][v]
+                            v_molecule = molecules[v][u]
+                            gout.add_edge(f"{u}-{u_molecule}",f"{v}-{v_molecule}",
+                                          n=gin[u][v]["n"])
+                        else:
+                            u_molecule = molecules[u][v]
+                            gout.add_edge(f"{u}-{u_molecule}", f"{v}",
+                                          n=gin[u][v]["n"])
+                # remove older nodes
+                gout.remove_nodes_from([u for u, _ in molecules.items()])
+            elif round == 1:
+                for (u, v), prop in gin.edges.items():
+                    # Skip singleton and cut vertices, which are excluded from the partition.
+                    if v not in molecules[u] or u not in molecules[v]:
+                        continue
+                    u_molecule = molecules[u][v]
+                    v_molecule = molecules[v][u]
+                    gout.add_edge(f"{u}_{u_molecule}", f"{v}_{v_molecule}", n=prop["n"])
 
-        num_singletons = Physlr.remove_singletons(gout)
-        print(
-            int(timeit.default_timer() - t0),
-            "Removed", num_singletons, "isolated vertices.", file=sys.stderr)
+            print(int(timeit.default_timer() - t0), "Separated molecules", file=sys.stderr)
+            num_singletons = Physlr.remove_singletons(gout)
+            print(
+                int(timeit.default_timer() - t0),
+                "Removed", num_singletons, "isolated vertices.", file=sys.stderr)
+            # gin.clear()
+            # gin = gout.copy()
+            gin = gout
+            round = round + 1
+            gout = 0
+            # Physlr.graph.clear()
+            # gout.clear()
+            # old_molecules = molecules
+            # gc.collect()
+
+        if 2>1:
+            # No effect on results
+            # only set to True to report the number of junctions after last round of mol-sep
+            Physlr.identify_junctions_graph(gin)
```

Comment:
I dont see the point to this code. Wouldn't it always print the number of junctions for whichever round that it is in?

Reply:
This code prints the number of junction "after" the last round. Normally we start printing number of junction "before" each round for rounds 2+. and takes some time to do so. So by disabling it we save some time for a report that is not necessary. I will just make it condition using a parameter with default value being `Fasle`.